test/container_app_test.py

```diff
@@ -58,12 +59,43 @@ async def test_container_snapshot_restore(container_client, tmpdir, servicer):
         encoding="utf-8",
     )
     with mock.patch.dict(
-        os.environ, {"MODAL_RESTORE_STATE_PATH": str(restore_path), "MODAL_SERVER_URL": servicer.container_addr}
+        os.environ, {"MODAL_RESTORE_STATE_PATH": str(restore_path), "MODAL_SERVER_URL": servicer.container_addr},
     ):
         io_manager.memory_snapshot()
         # In-memory Client instance should have update credentials, not old credentials
         assert old_client.credentials == ("ta-i-am-restored", "ts-i-am-restored")
 
+@pytest.mark.asyncio
+async def test_container_snapshot_restore_heartbeats(container_client, tmpdir, servicer):
+    io_manager = ContainerIOManager(api_pb2.ContainerArguments(), container_client)
+    restore_path = tmpdir.join("fake-restore-state.json")
+    restore_path.write_text(
+        json.dumps(
+            {
+                "task_id": "ta-i-am-restored",
+                "task_secret": "ts-i-am-restored",
+                "function_id": "fu-i-am-restored",
+            }
+        ),
+        encoding="utf-8",
+    )
+
+    asyncio.set_event_loop(asyncio.new_event_loop())
```

Comment:
This and the asyncio.sleep a few lines below were frivolous attempts at fixing CI.